incubator/command-dump/src/main/java/io/aklivity/zilla/runtime/command/dump/internal/airline/ZillaDumpCommand.java

```diff
@@ -0,0 +1,580 @@
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        final FrameFW frame = frameRO.wrap(buffer, index, index + length);
+        final long timestamp = frame.timestamp();
+
+        if (!nextTimestamp(timestamp))
+        {
+            return false;
+        }
+
+        switch (msgTypeId)
+        {
+        case BeginFW.TYPE_ID:
+            onBegin(beginRO.wrap(buffer, index, index + length));
+            break;
+        case DataFW.TYPE_ID:
+            onData(dataRO.wrap(buffer, index, index + length));
+            break;
+        case EndFW.TYPE_ID:
+            onEnd(endRO.wrap(buffer, index, index + length));
+            break;
+        case AbortFW.TYPE_ID:
+            onAbort(abortRO.wrap(buffer, index, index + length));
+            break;
+        case WindowFW.TYPE_ID:
+            onWindow(windowRO.wrap(buffer, index, index + length));
+            break;
+        case ResetFW.TYPE_ID:
+            onReset(resetRO.wrap(buffer, index, index + length));
+            break;
+        case ChallengeFW.TYPE_ID:
+            onChallenge(challengeRO.wrap(buffer, index, index + length));
+            break;
+        case SignalFW.TYPE_ID:
+            onSignal(signalRO.wrap(buffer, index, index + length));
+            break;
+        case FlushFW.TYPE_ID:
+            onFlush(flushRO.wrap(buffer, index, index + length));
+            break;
+        default:
+            //Should not happen
+            throw new IllegalStateException("Unexpected frame type: " + msgTypeId);
+        }
+        return true;
+    }
+
+    private boolean isStreamsFile(
+        Path path)
+    {
+        final int depth = path.getNameCount() - directory.getNameCount();
+        if (depth != 1 || !Files.isRegularFile(path))
+        {
+            return false;
+        }
+
+        final Matcher matcher = STREAMS_PATTERN.matcher(path.getName(path.getNameCount() - 1).toString());
+        return matcher.matches() && ((1L << parseInt(matcher.group(1))) & affinity) != 0L;
+    }
+
+    private void onDiscovered(
+        Path path)
+    {
+        if (verbose)
+        {
+            System.out.printf("Discovered: %s\n", path);
+        }
+    }
+
+    private boolean nextTimestamp(
+        final long timestamp)
+    {
+        if (timestamp != nextTimestamp)
+        {
+            nextTimestamp = Math.min(timestamp, nextTimestamp);
+            return false;
+        }
+        else
+        {
+            nextTimestamp = Long.MAX_VALUE;
+            return true;
+        }
+    }
+
+    private void onBegin(
+        BeginFW begin)
+    {
+        if (allowedBindings.isEmpty() || allowedBindings.contains(begin.routeId()))
+        {
+            final long streamId = begin.streamId();
+            TcpHeaderFW tcpHeader = createTcpHeader(streamId, Flag.SYN);
+            PcapPacketHeaderFW pcapHeader = createPcapPacketHeader(PSEUDO_ETHERNET_FRAME.length + tcpHeader.sizeof(),
```

Comment:
Is the IPv6 packet structure included in the `PSEUDO_ETHERNET_FRAME` bytes?
If so, how is the length conveyed to the IPv6 packet in the generated output?

Reply:
Logic changed, so it's not relevant anymore. IPv6 is modelled as flyweight.